Lib/unittest/mock.py

```diff
@@ -51,6 +51,13 @@ def _is_async_obj(obj):
         return False
 
 
+def _is_async_func(func):
+    if getattr(func, '__code__', None):
```

Comment:
Is it possible to have an Asunción callable class? If so this won’t work.

Would a check for ‘callable’ work here?

Reply:
I don't understand "Asunción callable class" Do you mean something like `async class Foo: pass` which I think is not supported. Please add an example of the same.

Reply:
There is no async version of `callable()` builtin like `acallable()`.
`asyncio.iscoroutinefunction()` works for functions only

Reply:
👍 note that the following will probably produce a standard mock rather than an async mock:

```python
class A:
   async def __call__(self):
       await asyncio.sleep(1)
```

We can cover this in a different PR I guess.

Reply:
Yes, it produces a MagicMock but there could be a class with a mix of sync and async dunder methods too so not sure of the behavior and how to detect classes like this. I would like to cover it as part of different PR since this contain fixes for `create_autospec`. Thanks for catching this.

```python
from unittest.mock import create_autospec

class A:

    async def __call__(self):
        await asyncio.sleep(1)

a = create_autospec(A)
b = a()
print(a)
print(b())
```

```
<MagicMock spec='A' id='4410501232'>
<MagicMock name='mock()()' id='4416843600'>
```

Reply:
Totally fair indeed.